### ISSUE TITLE
Future of vobject

### ISSUE THREAD

**Issue by pypingou** [adieu/vobject#5]
You seem to maintain the most followed/updated fork on vobject, are you willing to take over its maintainance (with help of course)?

I've been gathering patches from the different forks of your repo and made the sources pep8 compliant. Would you be interested in the pull-request?

**Comment by tBaxter** [comment]
Please, please take over the official project. I really dislike trying to install object as a dependency from github because the official pypi build is so outdated.

**Comment by pypingou** [comment]
Well updating pypi will remain a problem unless a) we manage to get the former maintainer to give us rights to upload or b) we rename the project (vobject3?)

**Comment by tBaxter** [comment]
My mistake. I thought you were the former maintainer.  I cannot find contact info for the former maintainer.

I would agree with renaming the project, although I would not suggest vobject3.

**Comment by Russell-Jones** [comment]
The PyPI maintainer is listed as Jeffrey Harris https://warehouse.python.org/project/vobject/ https://warehouse.python.org/user/jdharris/ 

His email address (in 2007, at least) http://osaf-selected-lists.3138168.n2.nabble.com/Feedback-on-Desktop-Search-Results-td3099110.html is listed here http://lists.osafoundation.org/pipermail/chandler-users/2007-October/000666.html (I'm reminded of "The Usual Suspects" for some reason).

He also appears to have a Linked in profile, in case you happen to use that service (I don't) https://www.linkedin.com/in/jeffreydavidharris

If you can't contact him and the project hasn't had any updates for a long time, it seems like it might be possible to take over the PyPI listing, see, e.g. http://sourceforge.net/p/pypi/support-requests/262/ (though that's inconclusive). There are a lot of other requests for transfer of ownership on that bug tracker, all those I've seen dealt with have been by Richard Jones, and I guess he's very busy as it took a while for a response to come through in some cases I saw, e.g. http://sourceforge.net/p/pypi/support-requests/218/ http://sourceforge.net/p/pypi/support-requests/343/ http://sourceforge.net/p/pypi/support-requests/365/ . It's not clear whether that's the best place to ask. The Python Package Index is a bit messy in some ways. I can't find any policy documents on or linked to on the PyPI site (though perhaps I'm not looking hard enough). I hear Guido has said that he has little or no interest in it, though there are efforts to clean it up http://www.youtube.com/watch?v=7An2GobbSWU

The original homepage is here, http://vobject.skyhouseconsulting.com/ and there is some discussion of a python 3 version (using 2to3, though personally I think 2.7 and 3.3+ compatability is a better approach) on the mailing list there http://lists.skyhouseconsulting.com/pipermail/vobject/2014-September/thread.html though the list is pretty quiet.

**Comment by Russell-Jones** [comment]
Also, Christian M. Amsüss http://christian.amsuess.com/ has apparently already written a Python 3 only port. See https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=747554 
In that bug report, Christian mentions that it can be checked out with
`git clone git://prometheus.amsuess.com/vobject -b python3`

**Comment by Russell-Jones** [comment]
Oh, and if anyone has a StackExchange account, perhaps help get this http://stackoverflow.com/questions/15371806/python3-library-for-icalendar-ics-files question re-opened or add some information about vobject to it? :)

**Comment by Russell-Jones** [comment]
Or, if you're just using iCal, consider using the icalendar module from Plone via PyPI. That's what django-scheduler just did (that former dependency is why I was interested in the status of vobject).

**Comment by pypingou** [comment]
@Russell-Jones unfortunately the icalendar module is not as feature full as vobject iirc. I played with it at one point and turned out that either it's not producing or not parsing some iCal file correctly (don't have the details fresh in mind though :-/)

**Comment by tBaxter** [comment]
I believe my fork is now functioning for 2.7 and 3.4. At the very least, a whole bunch of new unit tests are passing: https://github.com/tBaxter/vobject

That said, the vobject code is still some pretty crusty stuff, and this thing really needs someone to give it more attention. I do not want to be that person. I've spent too much time already getting my fork this far along.

**Comment by pypingou** [comment]
@tBaxter have you tried your fork against the latest version of dateutil ?

The latest release of vobject by the original authors breaks: http://jenkins.cloud.fedoraproject.org/job/fedocal/456/console

**Comment by tBaxter** [comment]
Correct. To work with python3, you need dateutil 2.0 or better, which in turn necessitates some changes in the vobject codebase. My fork is using 2.0: https://github.com/tBaxter/vobject/blob/master/setup.py#L59

**Comment by pypingou** [comment]
I just wanted to point out that there is a discussion going on about the future of vobject on the vobject mailing list: http://lists.skyhouseconsulting.com/pipermail/vobject/2015-January/thread.html

**Comment by hadleyrich** [comment]
The discussions seem to have stalled. I'd suggest that if you're interested in maintaining just set up a wiki page from your repo here and email the old maintainer. There's much more activity here on GitHub than SourceForge.

**Comment by tBaxter** [comment]
I would tend to agree, and think it's up to you and me, adieu.

I've got a major uplift to Python3 and a new test suite.
You've got a lot more people watching yours. 

Not sure how much either of have huge interest, but drop me a line and let's talk.

**Comment by adieu** [comment]
@tBaxter sorry for the late reply. I cloned the repo to fix some bugs for a project several years ago. Then I move on to other projects and never used vobject again. So I wouldn't be the right choice for the maintainer position.

I see you and others put much efforts to improve this package. Can I make my contribution by transfer the ownership of this repo which seems to has most stars and forks on github to the new maintainer? It looks to me the best way to keep this project maintained.

**Comment by tBaxter** [comment]
Thanks. I'm a pretty reluctant maintainer myself. I don't really understand
all of it, and don't have a real passion for it. I just don't want to see
it die, either.

Nevertheless, it looks like someone needs to pick it up, so I guess I'll do
it. I'll work on contacting the prior owner, setting up a new repo, putting
it on pypi, etc.

**Comment by tBaxter** [comment]
Well, I thought it was moving in April. Not seeing any movement I jumped in. I give you https://github.com/tBaxter/python-card-me, which is on pypi now.

**Comment by hadleyrich** [comment]
Good choice. Thank you for stepping up. I'll use and suggest that going forward.
